demo-app/js/app.js

```diff
@@ -42,6 +43,15 @@
     }
     listContainer.addEventListener('hidden-new-content', updateNewIndicator);
 
+    function openGaiaDialog(evt) {
+      var detail = evt.detail;
+      var li = list.getItemAt(detail.index);
+      var text = li.querySelector('h3').textContent;
+      dialog.textContent = text + ' item clicked!';
```

Comment:
not a huge deal, but instead of adding `getItemAt` to the list (which might get out of hand later, not sure), maybe we can start by just adding `getRecordAt` to the source (`js/bacon-source.js`) and just access `.title` / `.body` on it.